mail-filter/spamassassin/spamassassin-3.4.4-r5.ebuild

```diff
@@ -0,0 +1,319 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit perl-functions systemd toolchain-funcs
+
+MY_P="Mail-SpamAssassin-${PV//_/-}"
+S="${WORKDIR}/${MY_P}"
+DESCRIPTION="An extensible mail filter which can identify and tag spam"
+HOMEPAGE="https://spamassassin.apache.org/"
+SRC_URI="mirror://apache/spamassassin/source/${MY_P}.tar.bz2"
+
+LICENSE="Apache-2.0 GPL-2"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~ppc ~ppc64 ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux ~x86-macos"
+IUSE="berkdb cron ipv6 ldap libressl mysql postgres qmail sqlite ssl test"
+RESTRICT="!test? ( test )"
+
+# The Makefile.PL script checks for dependencies, but only fails if a
+# required (i.e. not optional) dependency is missing. We therefore
+# require most of the optional modules only at runtime.
+REQDEPEND="dev-lang/perl:=
+	dev-perl/HTML-Parser
+	dev-perl/Net-DNS
+	dev-perl/NetAddr-IP
+	virtual/perl-Archive-Tar
+	virtual/perl-Digest-SHA
+	virtual/perl-IO-Zlib
+	virtual/perl-Time-HiRes"
+
+# SpamAssassin doesn't use libwww-perl except as a fallback for when
+# curl/wget are missing, so we depend on one of those instead. Some
+# mirrors use https, so we need those utilities to support SSL.
+#
+# re2c is needed to compile the rules (sa-compile).
+#
+# We still need the old Digest-SHA1 because razor2 has not been ported
+# to Digest-SHA.
+OPTDEPEND="app-crypt/gnupg
+	dev-perl/BSD-Resource
+	dev-perl/Digest-SHA1
+	dev-perl/Encode-Detect
+	|| ( dev-perl/GeoIP2 dev-perl/Geo-IP )
+	dev-perl/HTTP-Date
+	dev-perl/Mail-DKIM
+	dev-perl/Mail-SPF
+	dev-perl/Net-Patricia
+	dev-perl/Net-CIDR-Lite
+	dev-util/re2c
+	|| ( net-misc/wget[ssl] net-misc/curl[ssl] )
+	virtual/perl-MIME-Base64
+	virtual/perl-Pod-Parser
+	berkdb? ( virtual/perl-DB_File )
+	ipv6? ( dev-perl/IO-Socket-INET6 )
+	ldap? ( dev-perl/perl-ldap )
+	mysql? (
+		dev-perl/DBI
+		dev-perl/DBD-mysql
+	)
+	postgres? (
+		dev-perl/DBI
+		dev-perl/DBD-Pg
+	)
+	sqlite? (
+		dev-perl/DBI
+		dev-perl/DBD-SQLite
+	)
+	ssl? ( dev-perl/IO-Socket-SSL )"
+
+DEPEND="
+	ssl? (
+		!libressl? ( dev-libs/openssl:0= )
+		libressl? ( dev-libs/libressl )
+	)"
+BDEPEND="${REQDEPEND}
```

Comment:
Are you sure you got this right? BDEPEND is stuff which needs to be executed on host building the package. To quote [devmanual](https://devmanual.gentoo.org/general-concepts/dependencies/):
> BDEPEND specifies dependencies applicable to CBUILD, i.e. programs that need to be executed during the build, e.g. virtual/pkgconfig. DEPEND specifies dependencies for CHOST, i.e. packages that need to be found on built system, e.g. libraries and headers. 

Please read https://dev.gentoo.org/~mgorny/articles/the-ultimate-guide-to-eapi-7.html#build-time-dependencies-split-into-depend-and-bdepend in addition.

So a typical example for BDEPEND would be app-text/asciidoc. I.e. a package which builds docs/man pages during src_compile and which calls asciidoc to do that. If we would add asciidoc only to DEPEND and a user would cross compile on amd64 for arm for example, the asciidoc binary wouldn't run because asciidoc was compiled for arm...

Reply:
That's exactly why it's a draft still: I'm not strictly confident about any of it.  :-)  I was planning seeing if anyone #gentoo-perl might take a look at it.  (That's why I created the PR as a draft.)  But your eyes are definitely welcome.

I _think_ `BDEPEND` is right.  Here's my thought process.

```
# The Makefile.PL script checks for dependencies, but only fails if a
# required (i.e. not optional) dependency is missing. We therefore
# require most of the optional modules only at runtime.
REQDEPEND="dev-lang/perl:=
    ...
```

1. So that suggests everything in `REQDEPEND` was necessary for building, as without them the builds will fail.
2. The SSL libraries were pulled out of `REQDEPEND` and placed in `DEPEND` per this previous conversations with kent/mjo: https://github.com/gentoo/gentoo/pull/13766#discussion_r354071883
3. `test?` then implies needing all the "`OPTDEPEND`"'s, since the tests run during the build.  (Right?)

I'm most shaky on the `acct-{user/group}/spamd` -- I don't think they're necessary until install, so I _think_ they're just `RDEPEND`.

Any good way to tests these, short of spinning up a fresh system and getting it setup for cross-compiling for ARM or something?  That seems like a lot of compiling -- anyone got an AMI or VM similar setup for this?